bundles/core/org.eclipse.smarthome.core.audio/src/main/java/org/eclipse/smarthome/core/audio/AudioManager.java

```diff
@@ -44,27 +45,68 @@
      */
     void play(AudioStream audioStream, String sinkId);
 
+    /**
+     * Plays the passed audio stream on the given sink.
+     *
+     * @param audioStream The audio stream to play
+     * @param sinkId The id of the audio sink to use or null
+     * @param volume The volume to be used or null
+     */
+    void play(AudioStream audioStream, String sinkId, PercentType volume);
+
     /**
      * Plays an audio file from the "sounds" folder using the default audio sink.
      *
+     * @param fileName The file from the "sounds" folder
      * @throws AudioException in case the file does not exist or cannot be opened
      */
     void playFile(String fileName) throws AudioException;
 
+    /**
+     * Plays an audio file with the given volume from the "sounds" folder using the default audio sink.
+     *
+     * @param fileName The file from the "sounds" folder
+     * @param volume The volume to be used or null
+     * @throws AudioException in case the file does not exist or cannot be opened
+     */
+    void playFile(String fileName, PercentType volume) throws AudioException;
```

Comment:
Similarly here

Reply:
I guess we have a NPE or `AudioException` here. The `fileName` will be extended with the path to the "sound" folder and directly passed to the constructor of a `File` object.

Reply:
For granularity, it is always a good idea to throw specific exception, that is, for null scenario - NPE or IAE, usw.